src/services/auth-api/auth-api.service.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  HttpMethod
+} from 'src/services/common'
+import {
+  UserCreatePayload,
+  UserSignInPayload,
+  SignResponse,
+  User
+} from 'src/common/types'
+import { Http } from 'src/services/http'
+import { ApiPath, AuthApiPath, ContentType } from 'src/services/common'
+
+type Constructor = {
+  http: Http
+  apiPrefix: string
+}
+
+class AuthApi {
+  #http: Http
+  #apiPrefix: string
+
+  constructor({ http, apiPrefix }: Constructor) {
+    this.#http = http
+    this.#apiPrefix = apiPrefix
+  }
+
+  public signUp(payload: UserCreatePayload): Promise<SignResponse> {
```

Comment:
We will not use signUp API in our application. We will redirect user to browser for this purpose